### ISSUE TITLE
mark_service sends SIGHUP but openrc-run still marks based on exit status

### ISSUE THREAD

**Issue by markhindley** [OpenRC/openrc#741]
Hi,

I have been trying to mark the status of a service from within start(). Minimal example:

start() {
  true
  mark_service_stopped
}

However the status remains started.

Looking at [src/mark_service/mark_service.c](https://github.com/OpenRC/openrc/blob/f4ece6c94512e1736d3418b4426b88bc799c720e/src/mark_service/mark_service.c#L54), it appears the intention is to send SIGHUP and that prevent setting the status based on the start() exit status.

Looking at src/openrc-run/openrc-run.c, the only place I can see the flag set by the SIGHUP handler is [L1295](https://github.com/OpenRC/openrc/blob/master/src/openrc-run/openrc-run.c#L1295) and that just aborts the option handling loop.

Or are the comments in mark_service.c obsolete and there is a different way to achieve this?

Thanks

Mark

**Comment by markhindley** [comment]
Hi,

I have looked at this some more. The openrc-run.8 manpage implies that mark_service_* should work within the initscript.

I have a patch that fixes the behaviour. Since SIGHUP is already handled and utilised, I have opted for SIGUSR1 for the mark_service_* handling.

Thanks

Mark

[When-manually-marking-service-status-use-SIGUSR1-to-skip-default.txt](https://github.com/user-attachments/files/17003106/When-manually-marking-service-status-use-SIGUSR1-to-skip-default.txt)
